openmp/libomptarget/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -130,6 +130,40 @@ Error iterateAgentMemoryPools(hsa_agent_t Agent, CallbackTy Cb) {
                        "Error in hsa_amd_agent_iterate_memory_pools: %s");
 }
 
+/// Dispatches an asynchronous memory copy
+/// Enables different SDMA engines for the dispatch in a round-robin fashion.
+Error asyncMemCopy(void *Dst, hsa_agent_t DstAgent, const void *Src,
+                   hsa_agent_t SrcAgent, size_t Size, uint32_t NumDepSignals,
+                   const hsa_signal_t *DepSignals,
+                   hsa_signal_t CompletionSignal) {
+  static BoolEnvar OMPX_UseMultipleSdmaEngines{
```

Comment:
This will emit a global constructor which we really don't want to have scattered around in the plugin. We need to initialize this in the AMDGPUDeviceTy or somewhere else.

Reply:
Thanks for pointing that out. I moved the Envar into the AMDGPUDeviceTy and have added a boolean to the StreamTy, so we do not need to access the device for the value.